tempesta_fw/str.h

```diff
@@ -216,39 +211,30 @@ typedef struct tfwstr_t {
 	};
 	struct sk_buff	*skb;
 	unsigned long	len;
+	unsigned int	nchunks;
+	unsigned short	flags;
 	unsigned char	eolen;
-	unsigned int	flags;
 } TfwStr;
 
-#define DEFINE_TFW_STR(name, val) TfwStr name = { .data = (val), NULL, \
-						  sizeof(val) - 1, 0 }
-#define TFW_STR_FROM(s)		((TfwStr){.data = (char*)s, NULL, strlen(s)})
+#define TFW_STR_STRING(val)	((TfwStr){.data = (val), NULL, sizeof(val) - 1, 0})
```

Comment:
The initialisation looks incomplete, only first 4 are explicitly initialised. Either initialise all of them or only `.data` and `.len`. Same apply for `TFW_STR_FROM_CSTR()`